### PR TITLE
feat(node): Add state of peer in PeerTracker

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: node/src/peer_tracker.rs [hunk 1]
```diff
@@ -1,56 +1,163 @@
...
+use libp2p::{identify, Multiaddr, PeerId};
 
+#[derive(Debug)]
 pub struct PeerTracker {
-    peers: DashSet<PeerId>,
+    peers: DashMap<PeerId, InnerState>,
+}
+
+#[derive(Debug)]
+struct InnerState {
```
Comment (zvolin) [inline node/src/peer_tracker.rs:14]: maybe some name like `PeerInfo` would do?

Comment (oblique) [reply node/src/peer_tracker.rs]: yes sounds much better

File: node/src/peer_tracker.rs [hunk 2]
```diff
@@ -1,56 +1,163 @@
...
 
-    pub fn remove(&self, peer: PeerId) {
-        self.peers.remove(&peer);
+    pub fn clear_addresses(&self, peer: PeerId) {
+        let mut state = self.get(peer);
+
+        state.addrs.clear();
+        state.addrs.shrink_to_fit();
+
+        // Downgrade state. Note that we do not downgrade a connected peer.
```
Comment (zvolin) [inline node/src/peer_tracker.rs:90]: Is it even okay to remove the addresses of connected peer, or maybe the one that we are connected through?

Comment (oblique) [reply node/src/peer_tracker.rs:90]: Not sure. I don't use this method yet

Comment (oblique) [reply node/src/peer_tracker.rs:90]: I also had an idea that the connected address can be stored in a different field as a primary address. But in general I'm not sure if we really need to keep track of the addresses

**Review by zvolin — APPROVED** [review]
looks good

**Pull request merged**